L1Trigger/L1TCaloLayer1/src/UCTGeometry.hh

```diff
@@ -0,0 +1,160 @@
+#ifndef UCTGeometry_hh
+#define UCTGeometry_hh
+
+// UCT Geometric mapping between EB+EE/HB+HE & HF TPGs to CPT7s
+// Access methods check if the mapping is correct
+// Access methods also allow conversion of TPG to CTP7 mapping
+// This class only has functions and defintions, no local memory
+
+/*
+    
+  Crate Eta mapping:
+
+  eta:      -5 ... -4 ... -3 ... -2 ... -1 ... 0 ... 1 ... 2 ...  3 ...  4 ...  5
+  absIEta:  41   ...  30(29)28     ...        1 1     ...       28(29)30   ... 41
+  region:   12 11 10  9  8  7  6  5  4 3 2 1  0 0 1 2  3  4   5  6 7 8 9 10 11 12
+
+  For EB/EE towers we have 8-bit ET + 1-bit finegrain input
+  For HB/HE towers we have 8-bit ET + 5-bit feature (TBD)
+  For HF only 8-bits ET per input tower
+
+  Each EB/HB+EE/HE regions are 4x4 == ~0.35 x 0.35 iEta x iPhi wide
+
+  Calo TPG eta index is 1 through 28 for EB/HB+EE/HE
+  Note that tower 29 is unused in trigger as it shadows portion of HE
+  Corresponding negative values are for negative eta
+
+  The uHTR HF eta index is 30-41 for 12 divisions in eta between 3-5
+  Note that eta=30-39 are 10-degree phi towers.  However, Layer2 wants the same
+  5-degree phi towers as in 1-28.  Therefore, we make artificial splitting for
+  output.  For eta=40,42 we take 20-degree phi tower and split four ways !
+
+  In the HF region size is proposed as 2x4 HF towers in etaxphi == 0.33 x 0.35,
+  where the phi division of x4 assumes above /2 and /4 splits.
+
+*/
+
+#include <utility>
+
+#define NCrates 3
+#define NCardsInCrate 6
+#define NRegionsInCard 7
+#define NEtaInRegion 4
+#define NPhiInRegion 4
+#define NPhiInCard NPhiInRegion // For convenience of Layer-2 we always have the same number of phi divisions
+
+#define HFEtaOffset NRegionsInCard * NEtaInRegion + 1
```

Comment:
a `#define` should always be parenthesised, e.g. 
```c++
#define HFEtaOffset (NRegionsInCard * NEtaInRegion + 1)
```
otherwise something like `HFEtaOffset * 2` will evaluate to `NRegionsInCard * NEtaInRegion + 1 * 2`, not `(NRegionsInCard * NEtaInRegion + 1) * 2`.

Just an other reason for using `constexpr` instead of `#define`.